server/src/main/java/io/crate/expression/scalar/ToCharFunction.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed to CRATE Technology GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.expression.scalar;
+
+import io.crate.data.Input;
+import io.crate.expression.symbol.Literal;
+import io.crate.metadata.NodeContext;
+import io.crate.metadata.Scalar;
+import io.crate.metadata.TransactionContext;
+import io.crate.metadata.functions.Signature;
+import io.crate.types.DataTypes;
+import io.crate.types.IntervalType;
+import org.joda.time.Period;
```

Comment:
is it absolutely necessary to use joda time? I think we were actually moving away from it in favour of java time.

Reply:
I wasn't sure - I was following what we've used in other scalar functions that involve intervals such as `IntervalArithmeticScalar` and `IntervalTimestampArithmeticScalar`. Totally happy to change it though

Reply:
Yes, it was used there because it seems that it was hard to go with java time in that case, but in I think the general rule we follow is going with java time if it is possible.

Reply:
Just looking into it a bit further - would it be okay to use Joda for the time being here? `IntervalType.sanitizeValue()` itself returns a Joda period, and as far as I can tell converting between a Joda Period and a Java Period isn't super straightforward. Given we also use the Joda PeriodFormat here, maybe it makes sense to stick with Joda in this instance?